lib/matplotlib/axes/_axes.py

```diff
@@ -1345,13 +1342,15 @@ def eventplot(self, positions, orientation='horizontal', lineoffsets=1,
         linewidths = np.asarray(linewidths)
 
         if len(lineoffsets) == 0:
-            lineoffsets = [None]
+            raise ValueError('lineoffsets cannot be empty')
         if len(linelengths) == 0:
-            linelengths = [None]
-        if len(linewidths) == 0:
-            lineoffsets = [None]
+            raise ValueError('linelengths cannot be empty')
+        if len(linestyles) == 0:
+            raise ValueError('linestyles cannot be empty')
         if len(linewidths) == 0:
-            lineoffsets = [None]
+            raise ValueError('linewidths cannot be empty')
+        if len(alpha) == 0:
+            raise ValueError('alpha cannot be empty')
```

Comment:
This is added compared to the discussion we had. For similar behavior and it will still break with unequal sizes below.

An option is to skip all these `len() == 0` checks, except for `colors`, and rely on the unequal size below. Fewer checks, but slightly worse error message.